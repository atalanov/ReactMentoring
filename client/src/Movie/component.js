// @flow
import React from 'react';
import MovieImage from '../MovieImage';
import MovieDescription from '../MovieDescription';

export default props => (
  <div>
    <MovieImage src={props.item.poster_path} />
    <MovieDescription
      jenre={props.item.genres.join(', ')}
      year={props.item.release_date}
      name={props.item.title}
    />
  </div>
);

