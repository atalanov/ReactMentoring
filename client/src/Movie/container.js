// @flow
import React from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import actions from '../actions/filter-action';
import { events } from '../constants/constants';
import Component from './component';

const mapStateToProps = state => ({
    movie: state.reducer.filterReducer.movie,
    pending: state.reducer.filterReducer.pending,
});

function mapDispatchToProps(dispatch) {
    return {
        filterActions: bindActionCreators(actions, dispatch),
    };
}

export class Movie extends React.Component {
    componentDidMount() {
        this.props.filterActions.getByID(this.props.match.params.id);
    }
    render() {
        console.log(this.props.movie);
        if (this.props.movie) {
            return (<Component item={this.props.movie} />);
        }
        if (this.props.pending) {
            return (<div>Pending... Please stand by... </div>);
        }
        return (<div />);
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(Movie);
