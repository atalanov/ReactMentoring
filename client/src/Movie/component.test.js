import React from 'react';
import { shallow, render } from 'enzyme';
import Movie from './component';

describe('Movie tests', () => {
    test('Component Movie renders correctly', () => {
        const item = {
            name: 'TestFilm',
            jenre: 'Horror',
            year: '2018',
            posterUrl: '/testurl',
        };
        const component = shallow(<Movie item={item} />);
        expect(component).toMatchSnapshot();
    });
});
