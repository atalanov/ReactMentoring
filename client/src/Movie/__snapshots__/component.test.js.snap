// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie tests Component Movie renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    item={
      Object {
        "genres": Array [
          "Horror",
        ],
        "poster_path": "/testurl",
        "release_date": "2018",
        "title": "TestFilm",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Unknown
          src="/testurl"
        />,
        <Unknown
          jenre="Horror"
          name="TestFilm"
          year="2018"
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "src": "/testurl",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "jenre": "Horror",
          "name": "TestFilm",
          "year": "2018",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Unknown
            src="/testurl"
          />,
          <Unknown
            jenre="Horror"
            name="TestFilm"
            year="2018"
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "src": "/testurl",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "jenre": "Horror",
            "name": "TestFilm",
            "year": "2018",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
